### PR TITLE
Add option to filter out words that have been translated to an empty string

### PR DESCRIPTION
**Pull request by mattiadg**
Possible fix to #213

### CODE REVIEW

**Comment by JackTemaki** [comment]
If this filtered output is a separate file, couldn't we just introduce the new output file without any parameter? @curufinwe @michelwi do we have conventions for this?

**Comment by michelwi** [comment]
Do we really need the filtered and non-filtered lexicon at the same time? I'd say if filtering is enabled, the single output of this job should be the filtered lexicon.

This way all consecutive jobs can just access `ApplyG2PModelJob.out_g2p_lexicon` without having to know which output is present or desired.
Otherwise the information "job should filter the lexicon" has also be presented to other jobs/recipes in the vicinity of the original job to select the correct output.

**Comment by mattiadg** [comment]
> Do we really need the filtered and non-filtered lexicon at the same time? I'd say if filtering is enabled, the single output of this job should be the filtered lexicon.
> 
> This way all consecutive jobs can just access `ApplyG2PModelJob.out_g2p_lexicon` without having to know which output is present or desired. Otherwise the information "job should filter the lexicon" has also be presented to other jobs/recipes in the vicinity of the original job to select the correct output.

I think you're right on this, I am going to fix it accordingly

**Review by michelwi — APPROVED** [review]

File: g2p/apply.py [hunk 1]
```diff
@@ -81,3 +88,14 @@ def run(self):
                     stdout=out,
                     stderr=err,
                 )
+
+    def filter(self):
+        fd_out, tmp_path = mkstemp(dir=".", text=True)
+        with open(self.out_g2p_lexicon, "r") as lex:
```
Comment (michelwi) [inline g2p/apply.py:94]:
```suggestion
        with uopen(self.out_g2p_lexicon, "rt") as lex:
```
to be consistent with the writing call above.?

**Review by JackTemaki — APPROVED** [review]

**Pull request merged**